src/DerivativeSignal.cpp

```diff
@@ -110,8 +111,15 @@ namespace geopm
                             GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
         }
         double time = m_time_sig->sample();
-        double signal = m_y_sig->sample();
-        return compute_next(m_history, m_derivative_num_fit, time, signal);
+        size_t history_size = m_history.size();
+        // Check if this is the first call to sample() since the last
+        // call to read_batch()
```

Comment:
```suggestion
        // Check if this is the first call ever to sample() (history_size == 0)
        // Or check if this is the first call to sample() since the last call to read_batch() (last element of history buffer does not match the sampled time)
```